File: my-folder/0345-reverse-vowels-of-a-string/solution.java
class Solution {
    public String reverseVowels(String s) {
        int left = 0;
        int right = s.length() - 1;
        char[] c = s.toCharArray();
        String vowels = "aeiouAEIOU";
        while (left < right) {
            while (left < right && vowels.indexOf(c[left]) == -1) {
                left++;
            } 
            while (left < right && vowels.indexOf(c[right]) == -1) {
                right--;
            }
            char temp = c[left];
            c[left] = c[right];
            c[right] = temp;
            left++;
            right--; 
        }
        return new String(c);
    }
}

File: my-folder/0345-reverse-vowels-of-a-string/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        hs = set()
        hs.add('a')
        hs.add('e')
        hs.add('i')
        hs.add('o')
        hs.add('u')
        hs.add('A')
        hs.add('E')
        hs.add('I')
        hs.add('O')
        hs.add('U')

        p1 = 0
        p2 = len(s) - 1

        s = list(s)

        while p1 < p2:
            while p1 < p2 and s[p1] not in hs:
                p1 += 1
            while p1 < p2 and s[p2] not in hs:
                p2 -= 1
            s[p1], s[p2] = s[p2], s[p1]
            p1 += 1
            p2 -= 1
        
        return "".join(s)
